src/collective/ifttt/actions/ifttt.py

```diff
@@ -7,136 +7,121 @@
+payload_options = SimpleVocabulary([
+    SimpleTerm(value=u'Description', title=_(u'Description of action')),
+    SimpleTerm(value=u'Username', title=_(u'Username of Editor')),
+    SimpleTerm(
+        value=u'Event start date/time', title=_(u'Event Start Date/Time')
+    ),
 ])
 
 
-class IIftttAction(Interface):
+class IIftttTriggerAction(Interface):
     """
         Definition of the configuration available for a  Ifttt action
     """
-    ifttt_name = schema.Text(
+    ifttt_name = schema.TextLine(
         title=_(u'IFTTT event name'),
         description=_(u'The IFTTT event name'),
         required=False,
     )
 
-    method = schema.Choice(
-        title=_(u'Call method'),
-        vocabulary=methods,
-    )
-
-    payload = schema.Text(
-        title=_(u'Payload'),
+    payload_option = schema.Choice(
+        title=_(u'Choose 2nd Payload'),
+        vocabulary=payload_options,
         description=_(
-            u'Whether the 3rd payload is Description, '
+            u'Choose whether the 3rd payload is Description, '
             u'Username of editor, or Event Start Date/time'
         ),
+    )
+    payload = schema.Text(
+        title=_(u'Payload'),
         required=False,
+        description=_(u'Provide input for chosen payload')
     )
 
 
-@implementer(IIftttAction, IRuleElementData)
-class IftttAction(SimpleItem):
+@implementer(IIftttTriggerAction, IRuleElementData)
+class IftttTriggerAction(SimpleItem):
     """
         The implementation of the action defined before
     """
 
     ifttt_name = u''
-    method = u''
+    payload_option = u''
     payload = u''
 
     element = 'plone.actions.Ifttt'
 
     @property
     def summary(self):
         return _(
-            u'${method} ${ifttt_name}',
-            mapping=dict(method=self.method, ifttt_name=self.ifttt_name),
+            u'${ifttt_name} ${payload_option}',
+            mapping=dict(
+                ifttt_name=self.ifttt_name,
+                payload_option=self.payload_option,
+                payload=self.payload,
+            ),
         )
 
 
-def interpolate(value, interpolator):
-    """Recursively interpolate supported values"""
-    if isinstance(value, unicode):
-        return interpolator(value).strip()
-    elif isinstance(value, list):
-        return [interpolate(v, interpolator) for v in value]
-    elif isinstance(value, dict):
-        return dict([(k, interpolate(v, interpolator))
-                     for k, v in value.items()])
-    return value
-
-
 EXECUTOR = ThreadPoolExecutor(max_workers=1)
 
 
 @implementer(IExecutable)
-@adapter(Interface, IIftttAction, Interface)
-class IftttActionExecutor(object):
+@adapter(Interface, IIftttTriggerAction, Interface)
+class IftttTriggerActionExecutor(object):
     """
         The executor for this action
     """
     timeout = 120
 
-    def __init__(self, context, element, event):
-        self.context = context
+    def __init__(self, element):
```

Comment:
@Shriyanshagro Why did you changed this signature? This violates action executor adapter contract shown in `@adapter(Interface, IIftttTriggerAction, Interface)` (those define types for the values for init) resulting

```
  Module plone.contentrules.rule.rule, line 46, in __call__
  Module zope.component._api, line 110, in getMultiAdapter
  Module zope.component._api, line 123, in queryMultiAdapter
  Module zope.interface.registry, line 359, in queryMultiAdapter
  Module zope.interface.adapter, line 552, in queryMultiAdapter
TypeError: __init__() takes exactly 2 arguments (4 given)
```